data/docs/eos-updater-flatpak-autoinstall.d.5

```diff
@@ -0,0 +1,122 @@
+.\"
+.IP "\fIFlatpak Ref Action\fP"
+.IX Flatpak Ref Action
+.\"
+A JSON object containing, at minimum, properties \fB"action"\fP,
+\fB"serial"\fP and optionally \fB"filters"\fP. Valid values for
+\fB"action"\fP are \fB"install"\fP, and \fB"uninstall"\fP, each having
+their own required properties as explained below. The only valid type
+for \fB"serial"\fP, is an integer, which must be monotonically
+increasing as new entries are appended to the file.
+.\"
+.IP "\fIThe "filters" entry\fP"
+.IX Action filters
+.\"
+A \fB"\fIFlatpak Ref Action\fP"\fP may have a \fB"filters"\fP entry which
+is a JSON object with properties \fB"architectures"\fP, \fB"~architectures"\fP,
+\fB"locales"\fP and \fB"~locales"\fP. Each filter property takes a JSON
+array of strings containing architectures or locales to apply to the filter.
+.\"
+Where a filter is prefixed with a tilde (~), the action will not be applied
+if the system architecture or locales matches any entry in the array. Otherwise,
+the action will only be applied if the system architectures or locales matches
+at least one entry in the array.
+.\"
+Where an action is filtered out and not applied, it will never be applied
+again, even if the property being filtered on changes. For instance, if the
+system locale changes, the action will not be applied on the next boot if
+it could have been applied were the locale the correct value when the action
+were first applied.
+.\"
+.IP "\fI"install" actions\fP"
+.IX Install action
+.\"
+Where the fB"\action\fP"\fP property of a "\fBFlatpak Ref Action\fP" entry is
+"\fB"install"\fP", the action will describe a flatpak that should be
+installed upon the next boot of the deployment in which the action was
+introduced. The entry must have the additional properties "\fB"app"\fP",
+"\fB"ref-kind"\fP" and "\fB"collection-id"\fP", which describe the
+flatpak app ID, whether the flatpak is a runtime or an app and the
+\fBPostree\fP(1) collection-id to install the app from. See
+\fBflatpak\fP(1) and \fPostree\fB(1) for more information on flatpak app IDs,
+ref-kinds and collection-ids generally. If the flatpak is already installed
+when the action is applied, \fBeos-updater-flatpak-installer\fP(8) will attempt
+to upgrade it.
+\."
+Note that flatpaks are not pulled from the collection-id upon booting into
+the deployment, they are instead downloaded by \fBeos-updater\fP(8) during its
+"fetch" step whilst it prepares the commit to be deployed.
+\."
+.IP "\fI"uninstall" actions\fP"
+.IX Uninstall action
+.\"
+Where the fB"\action\fP"\fP property of a "\fBFlatpak Ref Action\fP" entry is
+"\fB"unnistall"\fP", the action will describe a flatpak that should be
+uninstalled upon the next boot of the deployment in which the action was
+introduced. The entry must have the additional properties "\fB"app"\fP",
+"\fB"ref-kind"\fP". It is not an error if the flatpak is already uninstalled
+when the action is applied.
+\."
+.SH "SEE ALSO"
```

Comment:
It would be useful to include an `EXAMPLES` section with a short example JSON file in (could use the one from the §(Autoinstall files) section of the design doc).